climada/engine/impact_calc.py

```diff
@@ -129,6 +129,19 @@ def impact(self, save_mat=True, assign_centroids=True,
                 )
 
         impf_col = self.exposures.get_impf_column(self.hazard.haz_type)
+
+        # check for compability of impact function id between impact function set and exposure
+        if any(impf_id not in self.impfset.get_ids(haz_type=self.hazard.haz_type) for
+               impf_id in np.unique(self.exposures.gdf[impf_col].values)):
```

Comment:
_or_ maybe - 

```suggestion
        known_impact_functions = self.impfset.get_ids(haz_type=self.hazard.haz_type)
        if not all(self.exposures.gdf[impf_col].isin(known_impact_functions)):
```

which is possibly easier to read

Reply:
Yes that is more readable, I also included this. Thanks!